share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -218,6 +218,30 @@ macro(rocm_set_cpack_gen)
     endif()
 endmacro()
 
+macro(rocm_set_cpack_compression)
+    # Default to xz unless otherwise specified.
+    # The CPack default compression type is gzip, but xz packages are much smaller.
+    # Native packages have defaulted to xz since at least Ubuntu 14.04.
+    if(NOT CPACK_DEBIAN_COMPRESSION_TYPE)
+        if(DEFINED ENV{ROCM_DEBIAN_COMPRESSION_TYPE})
+            set(CPACK_DEBIAN_COMPRESSION_TYPE $ENV{ROCM_DEBIAN_COMPRESSION_TYPE})
+        else()
+            set(CPACK_DEBIAN_COMPRESSION_TYPE "xz")
+        endif()
+    endif()
+
+    # xz compression is slow, but can be parallelized with CMake 3.20 or later.
+    # The compression ratio is marginally affected. The speedup is subject to
+    # diminishing returns at higher thread counts.
+    if(NOT CPACK_THREADS)
+        if(DEFINED ENV{ROCM_PKGTHREADS})
+            set(CPACK_THREADS $ENV{ROCM_PKGTHREADS})
+        else()
+            set(CPACK_THREADS "8")
```

Comment:
Should we use `ProcessorCount` to set the number of threads?

Reply:
That would be `set(CPACK_THREADS "0")`. I also think that would be a better default, but I hadn't tested on a 128-core machine and wanted to pick something I knew was safe.

I'm not certain, but I think the concurrency is bounded by `(total size) / (block size)` anyway, because going beyond 8 threads made no noticeable difference in my tests on rocBLAS and rocSOLVER.

In any case, I'll stop being lazy and I'll test out "0" on a server with lots of cores. It's probably fine.

Reply:
The rocSOLVER package built in 12 seconds.